coffee-jpa/src/main/java/hu/icellmobilsoft/coffee/jpa/sql/batch/IJpaBatchService.java

```diff
@@ -0,0 +1,112 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 - 2024 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.jpa.sql.batch;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import hu.icellmobilsoft.coffee.jpa.sql.batch.enums.Status;
+import hu.icellmobilsoft.coffee.se.api.exception.BaseException;
+
+/**
+ * JPA batch műveletek.
+ *
+ * @author attila-kiss-ot
+ * @since 1.0.0
+ */
+public interface IJpaBatchService {

```

Comment:
Miert van a coffee-ban egy olyan interfesz aminek onamgaban nincs ertelme es az implementaciojat a Frappee adja. Nem lehetne az egesz pakk a frappee-ban? Akinek kelle a BatchService funkcio egyszeruen berakja a dependency koze. Igy a interfesz es az implementacio kulon eletet elhet a coffee verziojatol

Reply:
Jelenleg a coffee elég monolitikus, nagyon sok minden van benne, rengeteg függőséggel. Nem tudom, hogy mi az irány. 

Ha arra felé megyünk, hogy a coffee-ról mindent levágunk és minden egyes megoldást külön projektbe szervezünk, így külön életciklusra rakunk, akkor egyetértek, hogy nincs köze a batch funkcionalitásnak a coffee-hoz. (modularizáció)

Ha abba az irányba megyünk, hogy a coffee, egy "JakartaEE solution set" (doksiból), akkor nekem szimpatikusabb az az irány, hogy a coffee-ban van minden "JakartaEE solution", így a coffee egy össszefogó projekt. Viszont a coffee-ba nem rakunk olyan implementációkat, aminek JakartaEE implementációs (pl Hibernate) függősége van. Meghúznám ott a vonalat, hogy a coffee-nak csak API függősége van. Ha a "solution"-t le tudja implementálni csak az API-ra építve, akkor örülünk. Viszont "solution"-t adhat csak interface szinten (lásd IJpaBatchService), aminek nagyon minimál a függőség, így kb bárhova behúzható. És implementációt pedig elég a microservice (war) szinten választani (mondjuk egy frappee-hibernate behúzásával a war pom.xml-be). Így az implementációk csak nagyon high level szinten jelennek meg.

Reply:
Szerintem a BatchService onmaga egy olyan pontosan meghatarazott funkcio ami onmagaban a hibernate kiterjesztesnek tekintheto es lenyegeben csak hibernate fuggosege van. Hasznalhato lenne mind Spring ugy EE oldalon. Nem egy tracing, logger vagy olyan kozponti elem ami masnak a resze.
A jelenlegi IJpaBatchService csak egy interface ami az egesz coffeeban nincs hasznalva sehol igy felesleges koros fuggoseg alakul ki a
coffee(IJpaBatchService) - frappee(IJpaBatchService impl) - coffee(jpa, se reszek)
kozott. Itt azt is bele kell szamolni hogy ez a kombinacio nem lessz fent tarthato minden helyzetben.

A frappee-t eltudom ugy kepzelni hogy lenne neki tobb modulja:
frappe-hibernate-batch
frappee-ee10-batch
frappee-ee11-batch
frappee-spring-batch

Mind 1 kozponti hibernate modul indul ki es lehet hasznalni mindhol minden kombinacioban.
Az elkovetkezo EE11 a jelen esetben is komplikaciokat fog okozni

Reply:
Körkörös függőség nem alakul ki, a coffe nem dependál a frappee-ra csak fordítva... De vettem 👍 

A frappee-ee10-batch és frappee-ee11-batch modulokat nem igazán értem, hogy mi lenne bennük. A hibernate függőség már eleve meghatároz egy jakarta.persistence-api verziót.

Az interface-hez ragaszkodnék, hogy alacsony szinteken ne jelenjen meg a hibernate, mint implementáció függés. Az alábbi modul szerkezetre gondoltam:

```
frappee
  frappee-bom
  frappee-jpa-batch-api
    IJpaBatchService
  frappee-hibernate-batch (dep hibernate)
    HibernateBatchService implements IJpaBatchService
```